feat: improve and extend repository's readme (#120)
eigerco/pallet-move: README.md

# Move Pallet

A pallet for substrate based blockchains to enable the usage of smart contracts written in the Move-language.

## Overview

> Smart contracts can directly be implemented and executed as Move scripts or modularized in Move modules. Therefor, the pallet supports publishing of Move modules, and the execution of Move scripts to achieve this functionality. In case of larger projects, the pallet provides the publishing of a bundle (multiple Move modules).

Reviewer: ```suggestion
Smart contracts can directly be implemented and executed as Move scripts or modularized in Move modules. Therefore, the pallet supports publishing Move modules and the execution of Move scripts to achieve this functionality. In the case of larger projects, the pallet provides the publishing of a bundle (multiple Move modules).
```

Author: Mhh, not sure about the "Therefor", these are two different words... I think both can be used here, but I wanted to use that one... do you prefer "Therefore" in this case?

Rest is done

Reviewer: Oops, sorry! - my Grammarly reported that one. Feel free to ignore my comment and resolve this discussion

> For the execution of Move scripts and the publication of Move modules, the Move source code has to be compiled and serialized into bytecode. For this purpose the tool [`smove`](https://github.com/eigerco/smove) is provided. The tool also provides further helpful features for developing and working with the Move language and this pallet.

Reviewer: ```suggestion
For the execution of Move scripts and the publication of Move modules, the Move source code has to be compiled and serialized into bytecode. For this purpose, the tool [`smove`](https://github.com/eigerco/smove) is provided. The tool also provides further helpful features for developing and working with the Move language and this pallet.
```

Author: Done

> For more informations and learning more about how to work with this pallet, the following entry points are provided:

Reviewer: ```suggestion
For more information and to learn more about how to work with this pallet, the following entry points are provided:
```

Author: Done

- [Move Script Example](#move-script-example)
- [Pallet Configuration in a Substrate-Node](#pallet-configuration-in-a-substrate-node)
- [Quickstart Guide with a Template-Node](#substrate-node-with-move-pallet)
- [Benchmarking](#benchmarking)

## Move Script Example

> As a simplified example a car rental situation will be modeled, by implementing a small Move module and a Move script. For the sake of simplicity, assume that developer Bob already implemented a Move module `CarPool`, which offers the following public method:

Reviewer: ```suggestion
A basic sample of the Move module and the Move script is shown below.
```

Author: Done

```move
module DeveloperBob::CarPool {
    // ...

    /// Requests the rent of a car on the given day and between given daytime hours.
    /// Returns `true` in case of success, otherwise `false`.

>     public fun rent_car(account: &signer, day: u16, hr_from: u16, hr_to: u16, token_limit: u128): bool {

Reviewer: `hr_from` - hours from?

What about making it shorter - it's just a showcase, it doesn't necessarily make too much sense:
```suggestion
    public fun rent_car(account: &signer, day: u16, lease_time: u16, token_limit: u128): bool {
```

Author: Done

        // ...
    }

    // ...
}
```

> This module will be explained in detail in [our tutorial](TODO), if you are interested in. Bob already published this module and stored it in his own memory space on blockchain. Now we can use this published module by executing the following Move script:

Reviewer: Too much text for our simple readme - those that have no clue what the Move is about will can the book below
```suggestion
```

Reviewer: Done

```move
script {
    use DeveloperBob::CarPool;

    fun rent_car(who: signer, day: u16, hr_from: u8, hr_to: u8, token_limit: u128) {
        CarPool::rent_car(&signer, day, hr_from, hr_to, token_limit)
    }
}
```

> For a general overview and further details of the Move language, have a look in the [Move-Book](https://move-language.github.io/move/introduction.html).

Reviewer: ```suggestion
For a general overview and further details of the Move language, have a look at the [Move-Book](https://move-language.github.io/move/introduction.html).
```

Author: Done

## Pallet Configuration in a Substrate-Node

> The pallet's configuration is very short. Beside the regular `RuntimeEvent` and a predefined `WeightInfo`, you only have to tell the pallet about your `Currency` handler:

Reviewer: ```suggestion
The pallet's configuration is very short. Besides the regular `RuntimeEvent` and a predefined `WeightInfo`, you only have to tell the pallet about your `Currency` handler:
```

```rust
impl pallet_move::Config for Test {
    type Currency = Balances; // here pallet-balances is used
    type RuntimeEvent = RuntimeEvent;
    type WeightInfo = ();
}
```

> Like mentioned before, the pallet provides three extrinsic calls for publishing modules and bundles, and for execution of scripts:

Reviewer: ```suggestion
As mentioned before, the pallet provides three extrinsic calls for publishing modules and bundles and for the execution of scripts:
```

Author: Done

```

It will start the `node-template` on the local interface. You can change the default behavior by passing your own command when running the docker image. All available options are in the [node template](https://docs.substrate.io/reference/command-line-tools/node-template/) documentation.

## Benchmarking

> Benchmarking and updating weights should be done each time new extrinsic is added to the pallet (weights are used to estimate transaction fees). Weights are obligatory for extrinsics that are available for users.

Reviewer: ```suggestion
Benchmarking and updating weights should be done each time a new extrinsic is added to the pallet (weights are used to estimate transaction fees). Weights are obligatory for extrinsics that are available for users.
```

Author: Done

> To update weights simply run:

Reviewer: ```suggestion
To update weights, simply run:
```

Author: Done

```bash
./target/release/node-template benchmark pallet --chain dev --pallet pallet-move --steps=50 --repeat=20 --wasm-execution=compiled --output ../pallet-move/src/weights.rs --template ./.maintain/frame-weight-template.hbs --extrinsic '*'
```

> when being in the substrate-based node directory root. Assumption is made that the pallet is located under `../pallet-move` directory. Template for the weights is located under `./.maintain/frame-weight-template.hbs` directory and can be obtained from Substrate repository.

Reviewer: ```suggestion
when being in the substrate-based node directory root. The assumption is made that the pallet is located under the `../pallet-move` directory. The template for the weights is located under the `./.maintain/frame-weight-template.hbs` directory and can be obtained from the Substrate repository.
```

Author: Done
